### DISCUSSION TITLE
Cyclic replication when connectCluster is set to source-cluster in MM2

### DISCUSSION THREAD

**Discussion by lenin-joseph** [strimzi/strimzi-kafka-operator#9198] [Q&A]
Hi, 

I tried to deploy MM2(unidirectional) closer to the source cluster and pointed connectCluster to the source-cluster. MM2 came up fine. However, I started seeing cyclic replication of messages within the source cluster itself. Is there any way to avoid this without setting connectCluster to the target-cluster?

Is it mandatory that the connectCluster should always point to the target-cluster? if so, CRD specification may have to be updated to reflect the same. Currently, it says

 connectCluster       <string> -required-
     The cluster alias used for Kafka Connect. The alias must match a cluster in
     the list at `spec.clusters`.
     
Regards,
Lenin Joseph

**Comment by scholzj** [comment]
This has to be the target cluster, yes.

**Reply by scholzj** [reply]
I can open a PR to improve the description.

**Reply by scholzj** [reply]
I opened #9202

**Reply by lenin-joseph** [reply]
thanks @scholzj for your swift response.

**Reply by LaithS** [reply]
Hi @scholzj, to your understanding is this a limitation in Kafka’s MM2 implementation or is it due to strimzi’s realisation of the MM2 functionality?

**Reply by scholzj** [reply]
This is limitation of Kafka MM2. It currently provides only source connectors. So the Connect cluster needs to be connected to the target cluster in order to produce there. The original plan was to have sink connectors as well. That would allow us to have a source cluster there as well. That is why the Strimzi custom resource is designed as it is - to be able to support that. But it has so far not been implemented.
